code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -103,9 +103,12 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 		. += HEAT_CAPACITY_VACUUM //we want vacuums in turfs to have the same heat capacity as space
 
 /// Calculate moles
-/datum/gas_mixture/proc/total_moles()
+/datum/gas_mixture/proc/total_moles(gas_id = null)
 	var/cached_gases = gases
-	TOTAL_MOLES(cached_gases, .)
+	if(!gas_id)
+		TOTAL_MOLES(cached_gases, .)
+		return
+	TOTAL_MOLES_SPECIFIC(cached_gases, gas_id, .)
```

Comment:
Rather than reusing the proc can you make this a separate proc, it doesnt hurt readability much and this proc is used in some very hot locations.